build.sh

```diff
@@ -38,8 +38,12 @@ fail() {
   exit 1
 }
 
+# Check for secrets that should not be committed
 /bin/bash ./git-secrets-scan.sh
 
+# Verify dependencies before starting the build
+/bin/bash ./check-dependencies.sh
```

Comment:
Most of these prerequisites are only needed if you want to package (`pack.sh`) to multiple languages which is not what `build.sh` is doing. I am wondering if we should split those up to `build-prereqs` and `pack-prereqs` and only check for the packaging deps if you attempt packaging.

Reply:
Just saw that at least `dotnet` is required by the CLI unit tests...

So I think I am taking this comment back - let's just make sure the environment is fully ready.